### PR TITLE
fix(argo-workflows): add missing rules for workflow role to run HTTP templates

### PR DESCRIPTION
**Pull request by yu-croco**
This PR resolves https://github.com/argoproj/argo-helm/issues/1273 🙋 

---

Signed-off-by: yu-croco <yuki.kita22@gmail.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

File: charts/argo-workflows/templates/_helpers.tpl [hunk 1]
```diff
@@ -120,3 +120,19 @@ Return the target Kubernetes version
...
+{{- $tagVersion := (trimPrefix "v" .Chart.AppVersion) -}}
+{{- if (regexMatch "^v([0-9]\\.){2}([0-9]){1}.*" .Values.controller.image.tag) -}}
+    {{- $tagVersion = (regexFind "([0-9]\\.){2}([0-9]){1}" .Values.controller.image.tag) -}}
+{{- end -}}
+
+{{- if or (semverCompare ">=3.3.0" $tagVersion) (regexMatch "^latest.*" .Values.controller.image.tag) -}}
+    {{- print "workflowtasksets/status" }}
+{{- else -}}
+    {{- print "workflowtasksets" }}
+{{- end -}}
```
Comment (yu-croco) [inline charts/argo-workflows/templates/_helpers.tpl:137]: It was difficult to cover all of cases, but I think below is enough . :innocent:
- when
  - `.Values.controller.image.tag` is default (`""`)
  - `.Values.controller.image.tag` >= v3.3.0, v3.3.0-xxxxx
  - `.Values.controller.image.tag` == latest

```yaml
- apiGroups:
    - argoproj.io
  resources:
    - workflowtasksets/status
  verbs:
    - patch
```

- when
  - image < v3.3.0

```yaml
- apiGroups:
    - argoproj.io
  resources:
    - workflowtasksets
  verbs:
    - patch
```

File: charts/argo-workflows/templates/controller/workflow-role.yaml [hunk 2]
```diff
@@ -36,6 +36,19 @@ rules:
...
+      - workflowtasksets
+    verbs:
+      - list
+      - watch
+  - apiGroups:
+      - argoproj.io
+    resources:
+      - {{ include "argo-workflows.workflowRole.httpTemplatePatchResourceName" $ }}
+    verbs:
+      - patch
```
Comment (yu-croco) [inline charts/argo-workflows/templates/controller/workflow-role.yaml:51]: Rules are from https://github.com/argoproj/argo-workflows/blob/master/manifests/quick-start/base/agent-role.yaml 🙋

**Comment by vladlosev** [comment]
Why not simply include both?
```yaml
rules:
  - apiGroups:
      - argoproj.io
    resources:
      - workflowtasksets
    verbs:
      - list
      - watch
  - apiGroups:
      - argoproj.io
    resources:
      - workflowtasksets
      - workflowtasksets/status
    verbs:
      - patch
```
This avoids adding hard to understand logic. Eventually, when everyone moves over to later versions of Argo Workflows, the older permission can be removed. You can add a TODO comment to that end.

**Comment by yu-croco** [comment]
Thank you for reviewing!
I did not come up with it and you are right. 😇 

I fixed this on https://github.com/argoproj/argo-helm/pull/1291/commits/b6b367e720b352a02211a4373c0a9b33540447ef 🙋

**Review by vladlosev — APPROVED** [review]
LGTM

**Review by jmeridth — APPROVED** [review]

**Pull request merged**